### PR TITLE
fix: panic on bucket key buffer for Stats Group By CMD

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- The bucket key on the group by is paniced because the buffer allocated for the bucket key is not enough.
- In this PR, it will check the size being written and if that size is not enough then it will increment the buffer.

# Testing
- Tested by making a new unit test that checked whether the buffer size is being incremented according to column sizes.
- All unit and functional test cases pass.

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/query/processor/statscommand_test.go [hunk 1]
```diff
@@ -379,3 +379,40 @@ func Test_ProcessSegmentStats(t *testing.T) {
...
+
+	knownValues := getTestData()
+	processor := getGroupByProcessor()
+	assert.Equal(t, 0, len(processor.bucketKeyWorkingBuf))
+
+	col1Values := knownValues["col1"]
+	factorSize := 5
+
+	for i := 0; i < len(col1Values); i++ {
+		byteVal := make([]byte, utils.MAX_RECORD_SIZE*3)
```
Comment (AndrewHess) [inline pkg/segment/query/processor/statscommand_test.go:395]: Where is this 3 coming from? Is it based off the `factorSize`? Using `factorSize-1` could make that more clear

Also, is this `factorSize` arbitrary, or is it based on some other property of the test?

**Review by AndrewHess — COMMENTED** [review]
Looks pretty good

File: pkg/segment/query/processor/statscommand_test.go (hunk 1 above)
Comment (Macbeth98) [reply pkg/segment/query/processor/statscommand_test.go:395]: oh my bad, that 3 was supposed to be `factorSize` when I was testing I created the variable and set it to 5 but forgot to replace it here.

Comment (Macbeth98) [reply pkg/segment/query/processor/statscommand_test.go:395]: yes, factorSize is arbitary, but basically any value greater than the len(groupbyColumns). i will add that as a comment

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**